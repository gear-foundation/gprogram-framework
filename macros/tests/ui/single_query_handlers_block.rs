use gprogram_framework_macros::{gprogram, query_handlers};

#[gprogram]
mod gprogram {
    use super::*;

    query_handlers!(
        fn this(value: u32) -> Result<u32, ()> {
            Ok(value)
        }
    );

    query_handlers!(
        fn that(value: u32) -> Result<u32, ()> {
            Ok(value)
        }
    );
}

fn main() {}
