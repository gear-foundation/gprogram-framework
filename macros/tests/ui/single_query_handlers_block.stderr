error: There must be a single `query_handlers` block in the `gprogram` module
  --> tests/ui/single_query_handlers_block.rs:13:5
   |
13 |     query_handlers!(
   |     ^^^^^^^^^^^^^^

warning: unused import: `query_handlers`
 --> tests/ui/single_query_handlers_block.rs:1:43
  |
1 | use gprogram_framework_macros::{gprogram, query_handlers};
  |                                           ^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
